CHANGELOG_PENDING.md

```diff
@@ -1,5 +1,9 @@
 ### Improvements
 
+- [codegen/go] - Implement go type conversions for optional string, boolean, int, and float32
+  arguments, which should replace hoisted variable declarations in documentation. Fixes #8821.
+  [#8839](https://github.com/pulumi/pulumi/pull/8839)
```

Comment:
Nit: I would split this into two items: one - listing all new public functions (`Float64Ref` etc) and explaining how/when to use them; two - the program gen change (which is also user-facing because it affects imports).

Reply:
Will add in commit tomorrow morning